.github/workflows/JET.yml

```diff
@@ -0,0 +1,60 @@
+name: JET
```

Comment:
instead of a new workflow it should just be a new test group

Reply:
That won't be possible because we cannot add JET.jl to the Project.toml and test on julia pre-v1.12 and v1.12 due to package compat connstraints, see https://github.com/NumericalMathematics/DispersiveShallowWater.jl/pull/216#issuecomment-3218815131.

Reply:
No it just needs to allow JET v0.9 and v0.11

Reply:
You mean something like [2acf709](https://github.com/SciML/RecursiveArrayTools.jl/pull/506/commits/2acf709dd2ceb73257ca2b1c5c6f102bd76b4a8e)? Let's see if that works. Last time I tried to make something like this work, I always got compat problems.